### ISSUE TITLE
Describe how to use nonce/aud with VPs to mitigate replay attacks

### ISSUE THREAD

**Issue by awoie** [awoie/vp-token-spec#24]

**Comment by tlodderstedt** [comment]
Is replay detection a reason to limit our draft to VPs? I'm asking since VCs cannot be augmented with a tx specific nonce.

**Comment by awoie** [comment]
In my mental model, VCs don't have holder binding but @tplooker might disagree since I understood that derived BBS+ VC proofs might contain a linked secret-based holder binding in the VC itself rather than in the VP which would contain an empty proof property. IMO, this approach does not click very well with the `holder` property in the VP but I guess using the VC-proof rather the VP-proof might have some underlying mathematical reasons?

In the simple case, where domain/challenge (aud/nonce) would be only relevant for VPs, we could just use the `aud` and `nonce` from the OIDC request in the respective VP proof.

Why I think this is relevant for VPs only? In my mental model, the processing of the VP/VC will be passed on to a VP/VC processor and they typically rely on domain/challenge to verify a VP. For VCs, this is not required. Replay attacks for VCs is also not an issue as the VC is self-contained and it is just a static statement, it is not an authentication assertion. Note, the VC can contain a `jti` or `id` property which represents a nonce but this an issue the issuer has to deal with.  Also note that this paragraph might not apply to BBS+ with linked secrets where I hope Tobias can provide more clarity.

@AdamJLemmon @tplooker @Sakurann what do you think?

**Comment by tlodderstedt** [comment]
the apparent way is to map the nonce onto the challenge element of the vc data model.
Didn't find a definition yet. Is this field generally used for replay detection? Otherwise, people could use a leaked VP on another protocol to impersonate the legit holder.

**Comment by tlodderstedt** [comment]
I'm wondering whether vp_hash isn't sufficient. It at least ties the vp token to the id token, which is in turn bound to a transaction using nonce and/or PKCE. Replay with other services shall be prevented using audience restriction.

**Comment by awoie** [comment]
I agree but isn't `vp_hash` only OPTIONAL and only required if front channel is involved?

`challenge` / `nonce` and `domain` are specified by the [LD-Proof spec](https://w3c-ccg.github.io/ld-proofs/). In the spec, we could say that OP implementers SHOULD use the OIDC `aud` and `nonce` to protect VPs against replay attacks (without describing the actual mapping).
